Content.Shared/Chemistry/Solution/Solution.cs

```diff
@@ -43,6 +42,47 @@ public class Solution : IEnumerable<Solution.ReagentQuantity>, ISerializationHoo
         /// </summary>
         public Solution() { }
 
+        #region SolutionContainer Fields
+        /// <summary>
+        ///     If reactions will be checked for when adding reagents to the container.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField("canReact")]
+        public bool CanReact { get; set; } = true;
+
+        /// <summary>
+        ///     Volume needed to fill this container.
+        /// </summary>
+        [ViewVariables]
+        public ReagentUnit EmptyVolume => MaxVolume - CurrentVolume;
+
+        public ReagentUnit RefillSpaceAvailable => EmptyVolume;
+        public ReagentUnit InjectSpaceAvailable => EmptyVolume;
+        public ReagentUnit DrawAvailable => CurrentVolume;
+        public ReagentUnit DrainAvailable => CurrentVolume;
+
+        /// <summary>
+        ///     Checks if a solution can fit into the container.
+        /// </summary>
+        /// <param name="solution">The solution that is trying to be added.</param>
+        /// <returns>If the solution can be fully added.</returns>
+        public bool CanAddSolution(Solution solution)
+        {
+            return solution.TotalVolume <= EmptyVolume;
+        }
+
+        [DataField("maxSpillRefill")] public ReagentUnit MaxSpillRefill { get; set; }
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField("maxVol")]
+        public ReagentUnit MaxVolume { get; set; } = ReagentUnit.Zero;
+
+        [ViewVariables] public ReagentUnit CurrentVolume => TotalVolume;
+
+        [ViewVariables] public IEntity Owner { get; set; } = default!;
```

Comment:
Because this class is serialized if client code ever touched this it would explode because it would be null so ideally we'd remove it.

Reply:
If we don't pass these values in Solution a lot of methods will get IEntity as params. It looked really ugly to me.